### PR TITLE
add eslint config to prefer default export

### PR DESCRIPTION
**Pull request by aelnahas**
### PR Checklist:

Hey there! It's great to have you here. Here's a quick checklist to help you make a shiny PR:

* [ ] Have you checked out the guidelines in our [Contributing](../CONTRIBUTING.md) document?
* [ ] Have you looked if there might be other open [Pull Requests](../../../pulls) for the same update/change?

### Quick Description of the PR
- Reference ticket: https://github.com/FightPandemics/FightPandemics/issues/213
- Added eslint config to prefer default exports
- Fixed some errors that eslint caught.

What does it aim to achieve, and why?

- enforces the linting rule, along the way there were small errors, such as using un imported react hooks .

### An Overview of the Changes

* A few notes on your commits.
* To help understand them.

### CODE REVIEW

File: client/src/pages/NrMap.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React, { useState, useEffect, useRef } from "react";
+import React, { useState, useEffect, useRef, useCallback } from "react";
```
Comment (anniegiang) [inline client/src/pages/NrMap.js:1]: This seems out of scope for this PR ? 

@agmaal1 was working on this - can you double check if this is fine please?

Comment (aelnahas) [reply client/src/pages/NrMap.js:1]: @anniegiang I can revert it, only reason why I did that is when I ran the linter this came up as an error, `useCallback` is being used without an import, same with with render on the functional component

Comment (anniegiang) [reply client/src/pages/NrMap.js]: Ah ok, maybe we would fix that error in other PR.

File: client/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (anniegiang) [inline client/.eslintrc:1]: This [PR ](https://github.com/FightPandemics/FightPandemics/pull/258) configures ESlint in `client/.eslintrc.json`. We also have `backend/.eslintrc.json`. I think to stay consistent, maybe we could wait for that PR to merge, and then add these rules on top of that. What do you think?

Comment (robinv85) [reply client/.eslintrc:1]: Similar to my comment on the other ESLint changes for the frontend, shouldn't we use the create-react-app way to change ESLint config? https://github.com/FightPandemics/FightPandemics/pull/258#discussion_r422545352
> CRA isn't fully configurable unless "ejected", you can read more about this here: https://create-react-app.dev/docs/available-scripts#npm-run-eject but we probably want to avoid that unless absolutely necessary because it also has disadvantages.
> 
> For ESLint there is a setting that can be changed and then we can add some additional ESLint settings, that will be added to the settings that are provided by CRA, see https://create-react-app.dev/docs/advanced-configuration
> 
> So I think that's the way to go, and to stay within the scope of this PR you should probably only add what's needed, which is the `"sort-imports"` rule and `"sort-keys"`

Now I'm not sure how that works with CRA but indeed seems likely that it still needs to extend the `react-app` config.

Also didn't know that this needs the `import` plugin, or it might provide some additional benefits?

Comment (aelnahas) [reply client/.eslintrc:1]: @robinv85  I wasn't aware of that, I guess I could wait until the other PR is merged and then just apply this rule to it , and yea according to the guide was posted , it looked like that plugin was needed , cause the rule is part of that package.

**Comment by aelnahas** [comment]
@robinv85 I rebased from master and made the changes . Please take a look

**Comment by robinv85** [comment]
Before this is updated, I suggest we first update the config, I think this is probably best way to go:
- move Airtable keys to `config.js` (probably these will be removed eventually anyway
- add `client/.env` to git (remove from .gitignore)
- make sure the `EXTEND_ESLINT=true` setting is set for everyone

File: client/package.json [hunk 3]
```diff
@@ -56,6 +56,8 @@
     "@testing-library/react": "^10.0.3",
     "@testing-library/user-event": "^10.1.0",
     "dotenv": "^8.2.0",
+    "eslint": "^6.8.0",
```
Comment (robinv85) [inline client/package.json:59]: isn't this already included in create-react-app? Or does it need to be here if you also add a plugin?

**Comment by manualzuru** [comment]
@BMariscal @robinv85 @anniegiang @aelnahas should we merge this?

File: client/package.json (hunk 3 above)
Comment (robinv85) [reply client/package.json:59]: Following the other ESLint-related issues, I don't think this should be added in the package.json or it might have conflicting version with the one that is already included and it might cause problems

**Comment by idjevm** [comment]
Closing this, for now, feel free to reopen one this will be decided on. Also, the issue is still open so we can reopen this when the time comes.
